Some new thoughts on cutting a release, especially scripts/release.sh

daw3rd opened the pull request:
## Why are these changes needed?
We need to automate the process of version management and release cutting.  This is just a start with some initial thoughts based on recent discussion.  Initial changes include

1. Change .make.versions to define a common repo-wide version for all libraries and images.
2. Add scripts/release.sh which tries to outline the steps needed to changes versions, create and tag a branch.
3. Makefile new target to `show-version` based on .make.versions content
4. Fixed a lot of set-versions problems
5. Set all versions back to 0.2.0.dev6

Relies on `make build publish` being enough to actually publish the release to pypi and quay.io

Also see PR #297 

## Related issue number (if any).

#294

daw3rd committed:
Some new thoughts on cutting a release, especially scripts/release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

revit13 commented on transforms/universal/profiler/kfp_ray/Makefile:
```diff
@@ -29,8 +29,6 @@ image::
 
 load-image::
 
-set-versions: workflow-reconcile-requirements
```
Please rebase, this should be fixed now in dev branch. thanks

daw3rd replied:
done

roytman commented:
@daw3rd please do not do the release automation before we agree on the process. Some of the functionality can be done by GitHub actions

daw3rd committed:
Minor updates to release.sh script

Signed-off-by: David Wood <dawood@us.ibm.com>

n#

daw3rd committed:
Update repo.md for 2 make users and add workflow test of global make rules

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Update release.sh script for testing

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
more testing or release.sh script

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
More release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Revert to dev versions

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
release.sh

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
updated repo.md on makefiles

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Merge branch 'dev' into releasing

daw3rd committed:
add publish target to superworkflows Makefile

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
More release work

Added docs in .make.versions
Add RELEASE.md.
Fix checking for pypi credentials

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
add publish target to kind/Makefile

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd commented:
> @daw3rd please do not do the release automation before we agree on the process. Some of the functionality can be done by GitHub actions

I didn't think I was, except that I added the make tests for the test, clean, publish, build, set-versions top-level rules and some Makefiles needed to implement these

daw3rd committed:
install ../python for spark transforms in .make.defaults

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix kfp set versions, and noop versions

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix versioning in .make* and typo in noop/ray/pyproject.tmo

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Merge branch 'dev' into releasing

daw3rd committed:
comments and dont created empty pyproject.toml in .make.defaults

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
update release.sh to apply final version changes

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Version updates, version setting fixes, noop spark Dockerfile,

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Merge branch 'dev' into releasing

daw3rd committed:
set-versions inputcode2parquet'

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
restore INGEST_TO_PARQUET_VERSION removed in recent commit"

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
updated repo.md to point to include link to transform conventions

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix transform link added to repo.md in last commit

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Merge branch 'dev' into releasing

daw3rd committed:
fixed code2parquet versions from last merge and uninstall libs in better order in .make.defaults

Signed-off-by: David Wood <dawood@us.ibm.com>

revit13 commented on transforms/universal/profiler/ray/Makefile:
```diff
@@ -28,9 +28,8 @@ publish:: publish-dist publish-image
-# distribution versions is the same as image version.
 set-versions:
-	$(MAKE) TOML_VERSION=$(DOCKER_IMAGE_VERSION) .defaults.update-toml
+	$(MAKE) TRANSFORM_PYTHON_VERSION=$(DOCKER_IMAGE_VERSION) .transforms.set-versions
```
I do not think that's working , `TRANSFORM_PYTHON_VERSION` here is the value of `PROFILER_RAY_VERSION`  while it should be `PROFILER_PYTHON_VERSION`. I tried to fix that in https://github.com/IBM/data-prep-kit/pull/323

daw3rd replied:
You are correct.  To be more clear i set to `...=not-used` when there is a no python versions (profiler, e/fdedup, doc_id).  code_quality, malware and proglang_select did need the change.

HOWEVER, in the end since all version numbers are now the same, it was working ok :)

roytman commented:
We should deploy 2 kfp-data-processing images one for KFPv1 and another for KFPv2. Will try to update the script to proceed with that.

daw3rd committed:
Merge branch 'dev' into releasing

daw3rd committed:
remove redundant pip uninstall in .make.defaults

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix transform version settings when python transform code is referenced

Signed-off-by: David Wood <dawood@us.ibm.com>

revit13 commented on .make.versions:
```diff
@@ -28,43 +41,47 @@ else
-CODE_QUALITY_RAY_VERSION=0.4.0$(RELEASE_VERSION_SUFFIX)
+CODE_QUALITY_RAY_VERSION=$(DPK_VERSION)
+CODE_QUALITY_PYTHON_VERSION=$(DPK_VERSION)
+CODE_QUALITY_RAY_VERSION=$(DPK_VERSION)
```
It appears twice.

daw3rd committed:
.make.versions changes - remove redancy and unused macros, and define section for the repo's version dependencies.

Signed-off-by: David Wood <dawood@us.ibm.com>
